crates/router/src/connector/adyen/transformers.rs

```diff
@@ -289,139 +352,341 @@ impl TryFrom<&types::BrowserInformation> for AdyenBrowserInfo {
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for AdyenPaymentRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        let auth_type = AdyenAuthType::try_from(&item.connector_auth_type)?;
-        let reference = item.payment_id.to_string();
-        let amount = Amount {
-            currency: item.request.currency.to_string(),
-            value: item.request.amount,
-        };
-        let ccard = match item.request.payment_method_data {
-            api::PaymentMethod::Card(ref ccard) => Some(ccard),
-            api::PaymentMethod::BankTransfer
-            | api::PaymentMethod::Wallet(_)
-            | api::PaymentMethod::PayLater(_)
-            | api::PaymentMethod::Paypal => None,
-        };
+        match item.payment_method {
+            storage_models::enums::PaymentMethodType::Card => get_card_specific_payment_data(item),
+            storage_models::enums::PaymentMethodType::PayLater => {
+                get_paylater_specific_payment_data(item)
+            }
+            storage_models::enums::PaymentMethodType::Wallet => {
+                get_wallet_specific_payment_data(item)
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
 
-        let wallet_data = match item.request.payment_method_data {
-            api::PaymentMethod::Wallet(ref wallet_data) => Some(wallet_data),
-            _ => None,
-        };
+fn get_shopper_interaction_data(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> AdyenShopperInteraction {
+    match item.request.off_session {
+        Some(true) => AdyenShopperInteraction::ContinuedAuthentication,
+        _ => AdyenShopperInteraction::Ecommerce,
+    }
+}
 
-        let shopper_interaction = match item.request.off_session {
-            Some(true) => AdyenShopperInteraction::ContinuedAuthentication,
-            _ => AdyenShopperInteraction::Ecommerce,
-        };
+fn get_return_url(
```

Comment:
This need not be made into a function

Reply:
@Narayanbhat166 methods get_return_url, get_additional_data, get_amount_data, get_browser_info are being called at 3 places (get_card_specific_payment_data, get_wallet_specific_payment_data, get_paylater_specific_payment_data). These are repeated codes to do the same operation, making it inline will make the code duplicates.